src/generic/dirctrlg.cpp

```diff
@@ -420,6 +473,17 @@ bool wxGenericDirCtrl::Create(wxWindow *parent,
     if (m_filterListCtrl)
         m_filterListCtrl->FillFilterList(filter, defaultFilter);
 
+    // Did they ask for sorting by 
+    if ( style & (wxDIRCTRL_RCLICK_MENU           |
+                  wxDIRCTRL_EDIT_LABELS           |
+                  wxDIRCTRL_RCLICK_MENU_SORT_NAME |
+                  wxDIRCTRL_RCLICK_MENU_SORT_DATE ))
+    {
+        Connect(wxID_TREECTRL, wxEVT_COMMAND_TREE_ITEM_RIGHT_CLICK, wxTreeEventHandler(wxGenericDirCtrl::OnRightClick));
```

Comment:
Several remarks on this line:

* Please use `Bind()` in all new code.
* Please don't use compatibility `wxEVT_COMMAND_XXX` names.
* Please use `wxEVT_TREE_ITEM_MENU` instead of `RIGHT_CLICK`.